src/main/java/com/google/sps/servlets/NewRequestServlet.java

```diff
@@ -89,10 +90,25 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
           request.getParameter("end-time"),
           request.getParameter("max-stops"),
           userId);
+
+      if (request.getParameter("latitude") == null || request.getParameter("latitude") == null) {
+      // the user has set the address himself
+      slotManager.setSlotStartingPoint(request.getParameter("start-address"));
```

Comment:
Nit: indentation seems off in the diff view

Reply:
Done.